pkg/router/template/util/haproxytime/duration_parser.go

```diff
@@ -0,0 +1,75 @@
+package haproxytime
+
+import (
+	"errors"
+	"math"
+	"regexp"
+	"strconv"
+	"time"
+)
+
+var (
+	// OverflowError represents an overflow error from ParseDuration.
+	// OverflowError is returned if the value is greater than what time.ParseDuration
+	// allows (value must be representable as uint64, e.g. 9223372036854775807 nanoseconds).
+	OverflowError = errors.New("overflow")
+
+	// SyntaxError represents an error based on invalid input to ParseDuration.
+	SyntaxError = errors.New("invalid duration")
+
+	durationRE = regexp.MustCompile(`^([0-9]+)(us|ms|s|m|h|d)?$`)
+)
+
+// ParseDuration takes a string representing a duration in HAProxy's
+// specific format and converts it into a time.Duration value. The
+// string can include an optional unit suffix, such as "us", "ms",
+// "s", "m", "h", or "d". If no suffix is provided, milliseconds are
+// assumed. The function returns an OverflowError if the value exceeds
+// MaxTimeout, or a SyntaxError if the input string doesn't match the
```

Comment:
We moved the check for MaxTimeout to the caller so we should drop the "if the value exceeds MaxTimeout".

Reply:
ok